### PR TITLE
Mac OS: put version info in Info.plist

### PR DESCRIPTION
**Pull request by pvmm**
#145

### CODE REVIEW

File: build/version.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def _extractRevisionFromStdout(log, command, regex):
 
 def extractGitRevision(log):
 	return _extractRevisionFromStdout(
-		log, 'git describe --dirty', r'\S+?-(\S+)$'
+		log, 'git describe --always --dirty', r'\S+?-(\S+)$'
```
Comment (MBilderbeek) [inline build/version.py:40]: Why is this necessary? We don't need it in openMSX or Catapult either.

File: .github/workflows/workflow-build-macos.yaml [hunk 2]
```diff
@@ -16,6 +16,8 @@ jobs:
         git config --global init.defaultBranch master
     - name: Checkout repository code
       uses: actions/checkout@v3
+      with:
```
Comment (MBilderbeek) [inline .github/workflows/workflow-build-macos.yaml:19]: These fetch-depth changes do not seem to belong in this commit. Why are they included?

Comment (pvmm) [reply .github/workflows/workflow-build-macos.yaml:19]: Because git describe will generate `-none` version info without them.

File: build/version.py [hunk 3]
```diff
@@ -57,6 +57,8 @@ def extractRevision():
 	if not isdir('derived'):
 		makedirs('derived')
 	log = open('derived/version.log', 'w')
+	print('Extracting version info...', file = log)
+	print('Package version: %s' % packageVersion, file = log)
```
Comment (MBilderbeek) [inline build/version.py:61]: Perhaps we can bring this file (even) more in sync with the openMSX version in build/version.py?

File: build/version.py (hunk 1 above)
Comment (pvmm) [reply build/version.py:40]: It causes warnings on git when repository is shallow cloned.

Comment (pvmm) [reply build/version.py:40]: I think openMSX and Catapult should have it too btw.

Comment (MBilderbeek) [reply build/version.py:40]: I'd prefer to keep things in sync with openMSX and Catapult, if it's not strictly necessary, then I'd prefer to keep it the same. If it is, then we'd probably better change it in the other subprojects as well. Unfortunately, no one really proposed a way to keep the build system in sync between subprojects :(

Comment (pvmm) [reply build/version.py:40]: Can we apply it here then and then I make PRs for the other projects?

Comment (MBilderbeek) [reply build/version.py:40]: Fine with me. I guess @mthuurne has no objections either.

**Comment by sdsnatcher** [comment]
Complementary information: the lack of the version in the Info.plist file caused the debugger to not be found by the launchpad, so this is a welcome fix.

File: build/main.mk [hunk 4]
```diff
@@ -119,6 +119,10 @@ BINARY_FULL:=$(BINARY_PATH)/$(BINARY_FILE)
 VERSION_SCRIPT:=build/version2code.py
 VERSION_HEADER:=$(BUILD_PATH)/config/Version.ii
 GENERATED_HEADERS:=$(VERSION_HEADER)
+VERSION_LOG:=$(BUILD_BASE)/version.log
+VERSION_INFO=$(strip $(shell grep 'Package version:' $(VERSION_LOG) | awk -F':' '{print $$2}'))
+REVISION_INFO=$(strip $(shell grep 'Revision string:' $(VERSION_LOG) | awk -F':' '{print $$2}'))
+PACKAGE_DETAILED_VERSION=$(VERSION_INFO)-$(REVISION_INFO)
```
Comment (MBilderbeek) [inline build/main.mk:127]: Also build/main.mk was probably once copied from the openMSX version. Can you please check how the things you're trying to do here are done in openMSX, and if needed and possible, make it (more) similar?

**Comment by MBilderbeek** [comment]
That looks good! It's now quite similar to openMSX right? Or perhaps you can explain a bit what you did compared to last version.

**Comment by pvmm** [comment]
I used the same getDetailedVersion() from openMSX version.py to build the complete version info.

**Comment by MBilderbeek** [comment]
Thanks!
